### PR TITLE
Creates the ZD Target Docs. 

### PR DESCRIPTION
**Pull request by JeffNaef-Triggermesh**

### CODE REVIEW

File: docs/targets/zendesk.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Event target for Zendesk
+
+This event target recieves [CloudEvents][ce] and interfaces with Zendesk by either creating a new Ticket or placing a new Tag on a Ticket depending on the CloudEvent type.
```
Comment (cab105) [inline docs/targets/zendesk.md:3]: May want to ensure that the tag is applied to a pre-existing ticket.

File: docs/targets/zendesk.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Prerequisite: API token
+
+A Zendesk API key is required to utilize this target. The steps to obtain a key are outlined in the [Zendesk API Docs](https://support.zendesk.com/hc/en-us/articles/226022787-Generating-a-new-API-token-)
```
Comment (cab105) [inline docs/targets/zendesk.md:7]: May want to ensure whether your are using key or token, that you are consistent.  Or remove `API token` from the section header.

File: docs/targets/zendesk.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# Event target for Zendesk
```
Comment (cab105) [inline docs/targets/zendesk.md:1]: Please ensure consistent capitalizations.  Each word should be capitalized _except_ for prepositions.

File: docs/targets/zendesk.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+This event target recieves [CloudEvents][ce] and interfaces with Zendesk by either creating a new Ticket or placing a new Tag on a Ticket depending on the CloudEvent type.
+
+## Prerequisite: API token
+
+A Zendesk API key is required to utilize this target. The steps to obtain a key are outlined in the [Zendesk API Docs](https://support.zendesk.com/hc/en-us/articles/226022787-Generating-a-new-API-token-)
+
+## Deploying an instance of the Target
```
Comment (cab105) [inline docs/targets/zendesk.md:9]: Please ensure consistent capitalizations.  Each word should be capitalized _except_ for prepositions.

File: docs/targets/zendesk.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+Open the Bridge creation screen and add a target of type `Zendesk`.
+
+![Adding a Zendesk target](../images/zendesk-target/create-bridge-1.png)
+
+In the Target creation form, give a name to the event Target and add the following information:
```
Comment (cab105) [inline docs/targets/zendesk.md:15]: How about, "In the Target creation form, provide a name to the event Target, and add the following information"

File: docs/targets/zendesk.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+* **Email**: Email address associated with the Zendesk account.
+* [**Subdomain**][zd-subdom]: Name of the Zendesk subdomain, without the `zendesk.com` domain or `https://` scheme.
+* [**Token**][zd-token]: Reference to a [TriggerMesh secret][tm-secret] containing an API token to communicate with the Zendesk API, as described in the previous section.
+* **Subject**: An optional default subject assignment for new tickets
```
Comment (cab105) [inline docs/targets/zendesk.md:20]: You may want to highlight what the default subject would be if one is not provided.

File: docs/targets/zendesk.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+![Zendesk target form](../images/zendesk-target/create-bridge-2.png)
+
+After clicking the `Save` button, you will be taken back to the Bridge editor. Proceed to adding the remaining
```
Comment (cab105) [inline docs/targets/zendesk.md:24]: Replace `to` with `by`.  Also, could you rephrase to remove `you`?

File: docs/targets/zendesk.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+components to the Bridge, then submit it.
+
+![Bridge overview](../images/zendesk-target/create-bridge-3.png)
+
+A ready status on the main _Bridges_ page indicates that the Zendesk event target was
+successfully created and is ready to forward events into Zendesk.
+
+![Bridge status](../images/bridge-status-green.png)
+
+## Event types
```
Comment (cab105) [inline docs/targets/zendesk.md:34]: This should be `Event Types`

File: docs/targets/zendesk.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+A Zendesk Target will ONLY accept [CloudEvents][ce] with a "Type" of either `com.zendesk.ticket.create` OR `com.zendesk.tag.create`
+
+* Event's of type `com.zendesk.ticket.create` Expect both a `subject` and `body` to be preset.
```
Comment (cab105) [inline docs/targets/zendesk.md:38]: Events of type...

Comment (cab105) [reply docs/targets/zendesk.md:38]: Also, `Expect` should be `expect`

File: docs/targets/zendesk.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+  - **Example of type : `com.zendesk.ticket.create`**
```
Comment (cab105) [inline docs/targets/zendesk.md:39]: Do like that you are providing an example for each type.

File: docs/targets/zendesk.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+  - **Example of type : `com.zendesk.tag.create`**
+    ```sh
+    curl -v https://zendesktarget-triggermesh-zendesk.jnlasersolutions.dev.munu.io  \
+    -H "Content-Type: application/json" \
+    -H "Ce-Specversion: 1.0" \
+    -H "Ce-Type: com.zendesk.tag.create" \
+    -H "Ce-Source: some.origin/intance" \
+    -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+    -d '{"id":81 , "tag":"triggermesh"}'
+    ```
```
Comment (cab105) [inline docs/targets/zendesk.md:60]: Since the target returns an event, you should document what the response looks like so the user could potentially create an additional trigger/target to act on it.

File: docs/targets/zendesk.md (hunk 3 above)
Comment (antoineco) [reply docs/targets/zendesk.md:1]: For my own culture, why would we capitalize words that are not proper nouns? I rarely see that kind of casing. For example, you'll not see that in the AWS docs, or Zendesk docs, Microsoft docs, or any other docs I could think of.

If that's really a rule we want to stick to we should create a ticket to update all other docs pages as well.

File: docs/targets/zendesk.md (hunk 9 above)
Comment (antoineco) [inline docs/targets/zendesk.md:38]: Plural of event is events, not event's.

File: docs/targets/zendesk.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    curl -v https://zendesktarget-triggermesh-zendesk.jnlasersolutions.dev.munu.io  \
+    -H "Content-Type: application/json" \
+    -H "Ce-Specversion: 1.0" \
+    -H "Ce-Type: com.zendesk.ticket.create" \
+    -H "Ce-Source: some.origin/intance" \
+    -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+    -d '{"subject": "Hello", "body" : "World"}'
+    ```
+
+* Event's of type `com.zendesk.tag.create` Expect both a `id` and `tag` to be preset.
```
Comment (antoineco) [inline docs/targets/zendesk.md:50]: Plural of event is events, not event's.

File: docs/targets/zendesk.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+![Bridge overview](../images/zendesk-target/create-bridge-3.png)
+
+A ready status on the main _Bridges_ page indicates that the Zendesk event target was
+successfully created and is ready to forward events into Zendesk.
+
+![Bridge status](../images/bridge-status-green.png)
+
+## Event types
+
+A Zendesk Target will ONLY accept [CloudEvents][ce] with a "Type" of either `com.zendesk.ticket.create` OR `com.zendesk.tag.create`
```
Comment (antoineco) [inline docs/targets/zendesk.md:36]: Let's make that a list, since we will accept more sooner or later.
```suggestion
The Zendesk event target  can consume events of the following types:

* `com.zendesk.ticket.create`
* `com.zendesk.tag.create`
```

File: docs/targets/zendesk.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+* Event's of type `com.zendesk.ticket.create` Expect both a `subject` and `body` to be preset.
+  - **Example of type : `com.zendesk.ticket.create`**
+    ```sh
+    curl -v https://zendesktarget-triggermesh-zendesk.jnlasersolutions.dev.munu.io  \
```
Comment (antoineco) [inline docs/targets/zendesk.md:41]: IMO those docs are not a good place for throwing `curl` commands. If you want examples, please only post an example of payload, for example what `event-display` prints when it receives an event. Remember, those docs are for our users, not for developers.

File: docs/targets/zendesk.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+    -H "Ce-Type: com.zendesk.ticket.create" \
+    -H "Ce-Source: some.origin/intance" \
+    -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+    -d '{"subject": "Hello", "body" : "World"}'
+    ```
+
+* Event's of type `com.zendesk.tag.create` Expect both a `id` and `tag` to be preset.
+  - **Example of type : `com.zendesk.tag.create`**
+    ```sh
+    curl -v https://zendesktarget-triggermesh-zendesk.jnlasersolutions.dev.munu.io  \
```
Comment (antoineco) [inline docs/targets/zendesk.md:53]: IMO those docs are not a good place for throwing `curl` commands. If you want examples, please only post an example of payload, for example what `event-display` prints when it receives an event. Remember, those docs are for our users, not for developers.

File: docs/targets/index.md [hunk 16]
```diff
@@ -6,3 +6,4 @@ The following is a list of Triggermesh event Destinations known as Targets. Some
 
 * [AWS EventBridge](./awseventbridge.md): Forward arbitrary events to AWS EventBridge.
 * [Splunk](./splunk.md): Forward arbitrary events to Splunk.
+* [Zendesk](./zendesk.md): Forward arbitrary events to Zendesk
```
Comment (antoineco) [inline docs/targets/index.md:9]: Not true, should rephrase.

File: docs/targets/zendesk.md (hunk 7 above)
Comment (antoineco) [reply docs/targets/zendesk.md:24]: Can/should we really get around addressing the user directly in such doc? Other docs use that style, and I fail to see why this would be wrong.

File: docs/targets/zendesk.md (hunk 3 above)
Comment (cab105) [reply docs/targets/zendesk.md:1]: I'm reading this as a title, and thus under the MLA/AP style definition of titles: https://style.mla.org/capitalization-of-titles/

In this case, I'd expect `Event Target for Zendesk`

File: docs/targets/zendesk.md (hunk 7 above)
Comment (cab105) [reply docs/targets/zendesk.md:24]: For technical documentation, have been taught to use the 3rd person in order to remove the perception of bias and also remove a sense of familiarity.

File: docs/targets/zendesk.md (hunk 3 above)
Comment (antoineco) [reply docs/targets/zendesk.md:1]: OK thanks for the explanation. Created #33

File: docs/targets/zendesk.md (hunk 7 above)
Comment (antoineco) [reply docs/targets/zendesk.md:24]: Created #34

File: docs/targets/zendesk.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+    -H "Content-Type: application/json" \
+    -H "Ce-Specversion: 1.0" \
+    -H "Ce-Type: com.zendesk.ticket.create" \
+    -H "Ce-Source: some.origin/intance" \
+    -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+    -d '{"subject": "Hello", "body" : "World"}'
+    ```
+
+* Event's of type `com.zendesk.tag.create` Expect both a `id` and `tag` to be preset.
+  - **Example of type : `com.zendesk.tag.create`**
```
Comment (antoineco) [inline docs/targets/zendesk.md:51]: Bold is for emphasis, but this is not emphasis. I think what you're looking for is a subtitle here (`###`).
```suggestion
### Example of type `com.zendesk.tag.create`
```

File: docs/targets/zendesk.md (hunk 10 above)
Comment (antoineco) [inline docs/targets/zendesk.md:39]: Bold is for emphasis, but this is not emphasis. I think what you're looking for is a subtitle here (`###`).
```suggestion
### Example of type `com.zendesk.ticket.create`
```

File: docs/targets/index.md [hunk 18]
```diff
@@ -6,3 +6,4 @@ The following is a list of Triggermesh event Destinations known as Targets. Some
 
 * [AWS EventBridge](./awseventbridge.md): Forward arbitrary events to AWS EventBridge.
 * [Splunk](./splunk.md): Forward arbitrary events to Splunk.
+* [Zendesk](./zendesk.md): Forward events to Zendesk
```
Comment (antoineco) [inline docs/targets/index.md:9]: I would expect something more aligned with what the target does, like "Perform actions in Zendesk upon reception of certain events."

File: docs/targets/zendesk.md [hunk 19]
```diff
@@ -0,0 +1,69 @@
+# Event Target for Zendesk
+
+This event Target recieves [CloudEvents][ce] and interfaces with Zendesk by either creating a ticket or assigning a [tag][zd-tag] on a pre-existing ticket depending on the CloudEvent type.
```
Comment (antoineco) [inline docs/targets/zendesk.md:3]: Typo

File: docs/targets/zendesk.md (hunk 11 above)
Comment (JeffNaef-Triggermesh) [reply docs/targets/zendesk.md]: It only returns an ACK response but no data. -> https://github.com/triggermesh/knative-targets/blob/a20c39fb57814ba059d9b14a239f8bd0c0ea9856/zendesk/pkg/adapter/adapter.go#L180

```Host: zendesktarget-zendesktarget3.jeffthenaef.dev.munu.io
> User-Agent: curl/7.64.1
> Accept: */*
> Content-Type: application/json
> Ce-Specversion: 1.0
> Ce-Type: com.zendesk.tag.create
> Ce-Source: some.origin/intance
> Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79
> Content-Length: 30
> 
* We are completely uploaded and fine
* Connection state changed (MAX_CONCURRENT_STREAMS == 2147483647)!
< HTTP/2 500 
< content-length: 0
< date: Wed, 05 Aug 2020 20:00:56 GMT
< x-envoy-upstream-service-time: 2
< server: istio-envoy
< 
* Connection #0 to host zendesktarget-zendesktarget3.jeffthenaef.dev.munu.io left intact
* Closing connection 0

File: docs/targets/zendesk.md [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+
+## Deploying an Instance of the Target
+
+Open the Bridge creation screen and add a Target of type `Zendesk`.
+
+![Adding a Zendesk Target](../images/zendesk-target/create-bridge-1.png)
+
+In the Target creation form, provide a name to the event Target, and add the following information:
+
+* **Default Ticket Subject**: An optional ticket subject fallback if one is not provided in an incoming event
```
Comment (antoineco) [inline docs/targets/zendesk.md:17]: Need a dot at the end of the sentence.

File: docs/targets/zendesk.md [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+* **Zendesk Subdomain**: Name of the Zendesk [Subdomain][zd-subdom], without the `zendesk.com` domain or `https://` scheme.
+* **Zendesk Email**: Email address associated with the Zendesk account.
+* **Zendesk API Token**: Reference to a [TriggerMesh secret][tm-secret] containing a [token][zd-token] to communicate with the Zendesk API, as described in the previous section.
+* **Subject**: An optional default ticket subject assignment
```
Comment (antoineco) [inline docs/targets/zendesk.md:21]: 1. Missing dot.
2. I think this is a duplicate, you have two subject fields?

File: docs/targets/zendesk.md [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+
+For more information about using Zendesk, please refer to the [Zendesk documentation][docs].
+
+## Event Types
+
+A Zendesk event Target accepts the following [CloudEvent][ce] types:
+
+### com.zendesk.ticket.create
+
+Events of this type intened to create a new Zendesk ticket.
```
Comment (antoineco) [inline docs/targets/zendesk.md:41]: Typo

File: docs/targets/zendesk.md [hunk 23]
```diff
@@ -0,0 +1,69 @@
...
+
+This type expects a [JSON][ce-jsonformat] payload with the following properties:
+| Name  |  Type |  Comment |
+|---|---|---|
+| **subject**| string  |  The value of the subject field for this ticket |
+|  **body** |  string | The value of the body field for this ticket  |
+
+### com.zendesk.tag.create
+
+Events of this type intened to assign a [tag][zd-tag] to a pre-existing Zendesk ticket.
```
Comment (antoineco) [inline docs/targets/zendesk.md:51]: Typo

File: docs/targets/zendesk.md [hunk 24]
```diff
@@ -0,0 +1,69 @@
...
+
+[ce]: https://cloudevents.io/
+[ce-jsonformat]: https://github.com/cloudevents/spec/blob/v1.0/json-format.md
+[tm-secret]:https://docs.triggermesh.io/guides/secrets/
+
+[docs]: https://developer.zendesk.com/rest_api
+
+[zd-token]: https://support.zendesk.com/hc/en-us/articles/226022787-Generating-a-new-API-token-
+[zd-Target]: https://support.zendesk.com/hc/en-us/articles/203662136-Notifying-external-targets
+[zd-trigger]: https://support.zendesk.com/hc/en-us/articles/203662226-Triggers-resources
```
Comment (antoineco) [inline docs/targets/zendesk.md:67]: Unused.

File: docs/targets/zendesk.md (hunk 14 above)
Comment (JeffNaef-Triggermesh) [reply docs/targets/zendesk.md]: got it! I removed curl in favor of a property chart and included an example response of the `com.zendesk.ticket.create` event. Please let me know if the formatting is acceptable.

**Comment by JeffNaef-Triggermesh** [comment]
@cab105 I am **SO** blessed to have a grammar wizard around :) I believe all of your items are addressed now and I will apply them to the other targets. Thank you so much for your input!

File: docs/targets/zendesk.md (hunk 23 above)
Comment (JeffNaef-Triggermesh) [reply docs/targets/zendesk.md:82]: 😢 got it thank you

File: docs/targets/zendesk.md (hunk 21 above)
Comment (JeffNaef-Triggermesh) [reply docs/targets/zendesk.md:21]: yes it is!! sorry about that fixed!

File: docs/targets/zendesk.md [hunk 25]
```diff
@@ -1,68 +1,69 @@
...
 
 [ce]: https://cloudevents.io/
+[ce-jsonformat]: https://github.com/cloudevents/spec/blob/v1.0/json-format.md
+[tm-secret]:https://docs.triggermesh.io/guides/secrets/
+
+[docs]: https://developer.zendesk.com/rest_api
+
 [zd-token]: https://support.zendesk.com/hc/en-us/articles/226022787-Generating-a-new-API-token-
-[zd-target]: https://support.zendesk.com/hc/en-us/articles/203662136-Notifying-external-targets
+[zd-Target]: https://support.zendesk.com/hc/en-us/articles/203662136-Notifying-external-targets
```
[the remark this answers is not in the document]
Comment (JeffNaef-Triggermesh) [reply docs/targets/zendesk.md:97]: sorry this one and the other was copy-pasta from the source docs :)

File: docs/targets/zendesk.md (hunk 1 above)
Comment (sebgoa) [inline docs/targets/zendesk.md]:
```suggestion
This event target receives [CloudEvents][ce] and uses the Zendesk API to either create a new ticket or tag an existing one.
```

File: docs/targets/zendesk.md (hunk 4 above)
Comment (sebgoa) [inline docs/targets/zendesk.md]:
```suggestion
## Deploying an Instance of the Target
```

File: docs/targets/zendesk.md (hunk 8 above)
Comment (sebgoa) [inline docs/targets/zendesk.md]:
```suggestion
## Event Types
```

File: docs/targets/zendesk.md (hunk 9 above)
Comment (sebgoa) [inline docs/targets/zendesk.md]:
```suggestion
* Events of type `com.zendesk.ticket.create` expect both a `subject` and `body` to be set.
```

File: docs/targets/zendesk.md (hunk 12 above)
Comment (sebgoa) [inline docs/targets/zendesk.md]:
```suggestion
* Events of type `com.zendesk.tag.create` expect both a `id` and `tag` to be set.
```

File: docs/targets/zendesk.md [hunk 26]
```diff
@@ -0,0 +1,97 @@
...
+
+This event target receives [CloudEvents][ce] and uses the Zendesk API to either create a new ticket or tag an existing one.
+
+## Prerequisites
+
+A Zendesk API token is required to utilize this Target. The steps to obtain a token are outlined in the [Zendesk API Docs](https://support.zendesk.com/hc/en-us/articles/226022787-Generating-a-new-API-token-)
+
+## Deploying an Instance of the Target
+
+Open the Bridge creation screen and add a Target of type `Zendesk`.
```
Comment (antoineco) [inline docs/targets/zendesk.md:11]: @cab105 following your logic, this sentence is wrong because it uses the 2nd person. Mixing 3rd and 2ns persons should really be avoided.

Comment (cab105) [reply docs/targets/zendesk.md:11]: You are correct in that the mixing of personas should be avoided, and I take it you are referencing the implicit `you` in this case?

Comment (antoineco) [reply docs/targets/zendesk.md:11]: Yes, initiative would fall in the 2nd person right? Which means, if we rephrased other parts of the docs specifically to avoid the 2nd person, we should also get rid of all imperatives, correct?

Comment (sebgoa) [reply docs/targets/zendesk.md:11]: honestly, imperative and/or 2nd person is just fine.  How can we drive this to a merge so that we can also move on to the other 3 PRs and merge the frontend forms :)

Comment (sebgoa) [reply docs/targets/zendesk.md:11]: ?

Comment (antoineco) [reply docs/targets/zendesk.md:11]: It's approved already isn't it?

Comment (cab105) [reply docs/targets/zendesk.md:11]: If we wanted to scrub the implicit `you` in this case, then rephrasing it as "Begin by opening the bridge..."

However, given some of the comments left in the other PRs regarding the blending or allowing 2nd person I think we've established that we're okay with it. What I'd ask for is that we're consistent throughout.

Comment (antoineco) [reply docs/targets/zendesk.md:11]: Yeah, same here, consistent is the only thing I care about.

Comment (sebgoa) [reply docs/targets/zendesk.md:11]: ok, so let's use the 2nd person or imperative sentence consistenly. Do we agree ?

Comment (cab105) [reply docs/targets/zendesk.md:11]: Think the grammar police would agree.
